feat: eliminate RDS requirement

npflood opened the pull request:
The database adapter and parameters can be defined in the
node (custom json) if an RDS resource is not defined.
This is expected behaviour for Opsworks users.

BREAKING CHANGE: Sqlite3 is no longer set as the default database
adapter.

In order to use sqlite as the database adapter it must be defined
in the node.

npflood commented on libraries/helpers.rb:
```diff
@@ -1,4 +1,5 @@
 # frozen_string_literal: true
+# frozen_string_literal: true
```
This seems unnecessary. Not sure how this line got duplicated, but it is harmless

npflood commented on libraries/drivers_db_factory.rb:
```diff
@@ -1,10 +1,9 @@
 # frozen_string_literal: true
+require 'pry'
```
This is a mistake and should be removed. It is harmless, but unnecessary.

npflood commented on metadata.rb:
```diff
@@ -5,7 +5,7 @@
 description 'Set of chef recipes for OpsWorks based Ruby projects'
 long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
-version '0.2.1'
+version '0.3.0'
```
Bumped this to 0.3.0 because the change to removing sqlite as a default would be considered a breaking change. It seems too minor a change to wait for a full version revision.

npflood commented on spec/fixtures/node.rb:
```diff
@@ -11,14 +11,14 @@ def node(override = {})
+        #   host: 'dummy-project.c298jfowejf.us-west-2.rds.amazon.com',
+        #   database: 'dummydb',
+        #   reaping_frequency: 10
+        # },
```
The appropriate behavior for the definition of a database in the node should be to override the settings from RDS. As this is the case I commented out the database section. Ultimately the behavior of the scripts should implement this, and new tests should be written to simulate and verify that parameters can be overridden in this way.

npflood commented on spec/unit/recipes/configure_spec.rb:
```diff
@@ -258,9 +259,7 @@
-      db_config = Drivers::Db::Sqlite.new(
-        aws_opsworks_app(data_sources: []), dummy_node, rds: aws_opsworks_rds_db_instance(engine: 'sqlite3')
-      ).out
+      db_config = Drivers::Db::Sqlite.new(aws_opsworks_app(data_sources: []), dummy_node).out
```
It was a bit backward to fake an RDS to implement a Sqlite database, but understandable as the chef libraries were designed to only work with RDS.

ajgon commented:
Hey,

Thank you for this PR, I will take a look at it this evening, and comment/merge it.

npflood commented:
Anything I can do to help?

ajgon commented:
Hey, sorry for the delay.

I really like this PR. I have only three things:

* Have you tested it on live (OpsWorks) environment?
* Can you remove those `pry` and `frozen_string_literals` leftovers, and add this fix to the PR?
* Can you rename the commit from `feat eliminate....` to `feat: eliminate...` (add a colon)>

After that, I will be happy ti do the merge and deploy the new version to chef supermarket.

Thanks again!

npflood commented:
Happy to help! I've removed the leftovers and squashed the extra commits which fixed the missing colon in the commit message. 

I have tested this on my own sites, and haven't had an issue. I have not, however tested this with a deployment scenario that includes an RDS resource. I don't see a reason why the changes made should break that functionality, however it should be tried before pushing this out to Chef Supermarket. As I don't have any projects that use RDS with opsworks you can probably test this easier than I can.

ajgon commented:
Hey I tried it on my flows, with PgSQL and MySQL databases, and it seems to be working :). I'm merging the PR and deploying to supermarket.

Thanks again for this contribution!
